caikit/runtime/model_management/model_manager.py

```diff
@@ -252,33 +373,49 @@ def estimate_model_size(self, model_id, local_model_path, model_type) -> int:
         """
         return self.model_sizer.get_model_size(model_id, local_model_path, model_type)
 
-    def retrieve_model(self, model_id) -> ModuleBase:
+    def retrieve_model(self, model_id: str) -> ModuleBase:
         """Retrieve a model from the loaded model map by model ID.
 
         Args:
-            model_id(string): Model ID of the model to retrieve
+            model_id (str): Model ID of the model to retrieve
         Returns:
             response (caikit.core.module.ModuleBase):
                 A loaded Caikit model
         """
-        if not model_id:
+        if not model_id or not isinstance(model_id, str):
             raise CaikitRuntimeException(
                 StatusCode.INVALID_ARGUMENT, "Missing required model ID"
             )
 
-        # Now retrieve the model
-        if model_id not in self.loaded_models:
-            # We should not encounter this scenario, so if it happens, log
-            # it as an error-level log
+        # Now retrieve the model and fall back to lazy loading
```

Comment:
Here's the real guts of it all! This is where the lazy loading fallback happens.